modules/imgcodecs/test/test_jpeg.cpp

```diff
@@ -178,6 +178,40 @@ TEST(Imgcodecs_Jpeg, encode_decode_rst_jpeg)
     EXPECT_EQ(0, remove(output_normal.c_str()));
 }
 
+// See https://github.com/opencv/opencv/issues/25274
+TEST(Imgcodecs_Jpeg, regression25274_cmykjpeg)
+{
+    /*
+     * "test_1_c4.jpg" is CMYK-JPEG.
+     * $ convert test_1_c3.jpg -colorspace CMYK test_1_c4.jpg
+     * $ identify test_1_c4.jpg
+     * test_1_c4.jpg JPEG 480x640 480x640+0+0 8-bit CMYK 11240B 0.000u 0:00.000
+     */
+
+    cvtest::TS& ts = *cvtest::TS::ptr();
+    string input = string(ts.get_data_path()) + "readwrite/test_1_c4.jpg";
+    {
+        cv::Mat img = cv::imread(input);
+        ASSERT_FALSE(img.empty());
```

Comment:
The check is very weak. I propose to use at least countNonZero to check the image content. Original image in non-CMYK format is the best for reference.

Reply:
Thank you for your comment !
Comparing decoding result between RGB JPEG and CMYK JPEG is good.
`countNonZero()` requests single channel image, So I use `cvtest::norm()` instead of it.